### PR TITLE
Midea Climate support

### PR DESCRIPTION
**Pull request by dudanov**
## Description:

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1328

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by glmnet — CHANGES_REQUESTED** [review]
Is there a reason to have two docs file? I.e. is a component useful without the other?

Otherwise please merge them in a sine file were it will be easier to go thru the docs

**Comment by dudanov** [comment]
> Is there a reason to have two docs file? I.e. is a component useful without the other?
> 
> Otherwise please merge them in a sine file were it will be easier to go thru the docs

The idea is to separate the transport component (midea-dongle) common to all Midea devices ("usb" uart stick) and the components of various devices (currently supported air conditioners (midea-ac). It is known that there are other devices of this company, similarly controlled (humidifiers, water heaters, etc.)

**Comment by glmnet** [comment]
Well, in that case merging the docs will be preferred as there is no need to separate docs, you can even make the dongle component be auto loaded so you only have to configure the climate component itself, this makes a cleaner yaml file, you can still add the `midea_dongle:` element explicitly if you want to override some setting or the uart used, etc.

File: components/climate/midea_ac.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  - All other options from :ref:`Sensor <config-sensor>`.
+- **humidity_setpoint** (*Optional*): The information for the humidity indoor
+  sensor (experimental).
+
+  - **name** (**Required**, string): The name of the sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
+- **beeper** (*Optional*, bool, default: **false**): Beeper feedback on command.
+- **swing_horizontal** (*Optional*, bool, default: **false**): Enable **swing horizontal** option.
+- **swing_both** (*Optional*, bool, default: **false**): Enable **swing both** option.
```
Comment (glmnet) [inline components/climate/midea_ac.rst]: This is not how we document defaults, see other components for examples.
Usually a statement at the end like: "Defaults to ``false``."

**Review by glmnet — APPROVED** [review]

File: components/climate/midea_ac.rst (hunk 1 above)
Comment (dudanov) [reply components/climate/midea_ac.rst]: Done.

**Pull request merged**